### PR TITLE
fix(api): Add date filtering support to org group index

### PR DESCRIPTION
**Pull request by macqueen**
fixes https://getsentry.atlassian.net/browse/APP-995

going to add some more thorough tests tomorrow

### CODE REVIEW

File: src/sentry/api/endpoints/organization_group_index.py [hunk 1]
```diff
@@ -99,6 +100,12 @@ def get(self, request, organization):
         if not project_ids:
             return Response([])
 
+        try:
+            start, end = get_date_range_from_params(request.GET)
+        except InvalidParams as exc:
+            return Response({'detail': exc.message}, status=400)
+
+        # TODO(jess): check these event / short id results against date filter
```
Comment (macqueen) [inline src/sentry/api/endpoints/organization_group_index.py]: will do this tomorrow, either in this pr or in a follow up -- just want to unblock frontend work asap

Comment (markstory) [reply src/sentry/api/endpoints/organization_group_index.py]: If we get a short id and date options, I think the short ID should take priority.

**Review by wedamija — APPROVED** [review]

File: src/sentry/api/endpoints/organization_group_index.py (hunk 1 above)
Comment (macqueen) [reply src/sentry/api/endpoints/organization_group_index.py]: works for me. same for event id or do you think that should respect date filters?

Comment (markstory) [reply src/sentry/api/endpoints/organization_group_index.py]: I would generally favour the more precise input. event id and short-id are very specific keys and should take precedence over range selectors imo.

**Pull request merged**